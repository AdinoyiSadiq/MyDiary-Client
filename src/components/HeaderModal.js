import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { signout } from '../actions/authActions';

class HeaderModal extends Component {
  render() {
    const { showModal, signout } = this.props;
    return (
      <ul className={`headerModal ${showModal ? 'displayHeaderModal' : null}`}>
        <li><Link to="/main/profile">View Profile</Link></li>
        <li onClick={signout}>
          <div>
            SignOut
          </div>
        </li>
      </ul>
      );
  }
  
};

export default connect(null, { signout })(HeaderModal);