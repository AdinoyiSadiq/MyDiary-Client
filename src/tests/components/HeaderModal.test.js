import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import Root from '../../root';
import HeaderModal from '../../components/HeaderModal';

let wrapped;
let wrapped1;

beforeEach(() => {
  wrapped = mount(
    <Root>
      <MemoryRouter initialEntries={[{ key: 'testKey' }]}>
        <HeaderModal />
      </MemoryRouter>
    </Root>,
  );
  wrapped1 = mount(
    <Root>
      <MemoryRouter initialEntries={[{ key: 'testKey' }]}>
        <HeaderModal showModal={true} />
      </MemoryRouter>
    </Root>,
  );
});

afterEach(() => wrapped.unmount());


describe('HeaderModal UI', () => {
  describe('render features', () => {
    test('view should render as expected', () => {
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });
    test('view should render as expected when showModal class is used', () => {
      const tree = toJson(wrapped1);
      expect(tree).toMatchSnapshot();
    });
  });
});