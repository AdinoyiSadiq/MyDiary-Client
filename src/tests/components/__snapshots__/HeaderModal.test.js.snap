// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderModal UI render features view should render as expected 1`] = `
<_default>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <MemoryRouter
      initialEntries={
        Array [
          Object {
            "key": "testKey",
          },
        ]
      }
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "key": "testKey",
                "pathname": "/",
                "search": "",
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/",
              "search": "",
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Connect(HeaderModal)>
          <HeaderModal
            signout={[Function]}
          >
            <ul
              className="headerModal null"
            >
              <li>
                <Link
                  replace={false}
                  to="/main/profile"
                >
                  <a
                    href="/main/profile"
                    onClick={[Function]}
                  >
                    View Profile
                  </a>
                </Link>
              </li>
              <li
                onClick={[Function]}
              >
                <div>
                  SignOut
                </div>
              </li>
            </ul>
          </HeaderModal>
        </Connect(HeaderModal)>
      </Router>
    </MemoryRouter>
  </Provider>
</_default>
`;

exports[`HeaderModal UI render features view should render as expected when showModal class is used 1`] = `
<_default>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <MemoryRouter
      initialEntries={
        Array [
          Object {
            "key": "testKey",
          },
        ]
      }
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "key": "testKey",
                "pathname": "/",
                "search": "",
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/",
              "search": "",
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Connect(HeaderModal)
          showModal={true}
        >
          <HeaderModal
            showModal={true}
            signout={[Function]}
          >
            <ul
              className="headerModal displayHeaderModal"
            >
              <li>
                <Link
                  replace={false}
                  to="/main/profile"
                >
                  <a
                    href="/main/profile"
                    onClick={[Function]}
                  >
                    View Profile
                  </a>
                </Link>
              </li>
              <li
                onClick={[Function]}
              >
                <div>
                  SignOut
                </div>
              </li>
            </ul>
          </HeaderModal>
        </Connect(HeaderModal)>
      </Router>
    </MemoryRouter>
  </Provider>
</_default>
`;
